apps/mobile/src/screens/SettingsScreen/NotificationSettingsScreen.tsx

```diff
@@ -55,11 +121,186 @@ export function NotificationSettingsScreen() {
     [pushNotificationPermissions]
   );
 
+  const userEmail = query?.viewer?.email?.email;
+
+  const isEmailUnverified = useMemo(() => {
+    return DISABLED_TOGGLE_BY_EMAIL_STATUS.includes(query?.viewer?.email?.verificationStatus ?? '');
+  }, [query]);
+
+  const shouldShowEmailSettings = useMemo(
+    () => userEmail && !isEmailUnverified,
+    [userEmail, isEmailUnverified]
+  );
+
+  const currentEmailNotificationSettings = query?.viewer?.email?.emailNotificationSettings;
+
+  const [emailSettings, setEmailSettings] = useState<EmailNotificationSettings>({
```

Comment:
so i think we can definitely share a lot of this business logic across mobile <> web!

can use something like
```js
export function useEmailNotificationSettings() {
  const [emailSettings, setEmailSettings] = useState...
  
  const computeIsToggleChecked = ... // small rename here since it's a func

  const updateEmailNotificationSettings = useUpdateEmailNotificationSettings()

  const handleToggle = async ...

  return useMemo(() => ({
    emailSettings,
    computeToggleChecked,
    updateEmailNotificatinSettings,
    handleToggle,
    ...
  }), [...])
}
```

as you play with it you'll start to see what can be self-contained within the hook vs. returned out of the hook for access

Reply:
for the areas that are slightly different for mobile vs. web, such as displaying a toast, that can be passed in as a callback to the function from each platform.

e.g.
```js
const handleToggle = useCallback(async (settingType: string, settingTitle: string, onSuccess, onFailure) => {
  ...
  try {
     ...
     onSuccess()
  } catch (e) {
     onFailure(e)
  }
```

let's also make sure that if we're passing in a bunch of variables into `handleToggle`, we pass in a single object for more explicit parameterization

Reply:
got ittt, made the changes now!